# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveUtil.h
# Commit: 729990f9affb
# Full Hash: 729990f9affbcbc05925bf1cbc5749016664bc3f
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-09-09 20:57:47
# Regressor Bug: 1979588
# File Overlap Count: 2
# Overlapping Files: dom/base/test/test_text-fragments-create-text-directive.html, dom/base/TextDirectiveCreator.cpp
# Description:
#   Bug 1979588 - Text Fragments: Correctly deal with long words at end of range for range-based matching. r=dom-core,smaug
#   
#   The range-based approach for creating a text directive uses the word breaker algorithm
#   to identify where to split between `start` and `end` term. If the range ends with a word
#   which is over 50% of the length content, the algorithm would return wordEnd=length,
# ==============================================================================

diff -r 12d0bf6a782e -r 729990f9affb dom/base/TextDirectiveUtil.h
--- a/dom/base/TextDirectiveUtil.h	Tue Sep 09 14:32:24 2025 +0000
+++ b/dom/base/TextDirectiveUtil.h	Tue Sep 09 14:35:37 2025 +0000
@@ -551,8 +551,10 @@
     return 0;
   }
 
-  MOZ_ASSERT(!nsContentUtils::IsHTMLWhitespace(aReferenceString.First()));
-  MOZ_ASSERT(!nsContentUtils::IsHTMLWhitespace(aReferenceString.Last()));
+  MOZ_DIAGNOSTIC_ASSERT(
+      !nsContentUtils::IsHTMLWhitespace(aReferenceString.First()));
+  MOZ_DIAGNOSTIC_ASSERT(
+      !nsContentUtils::IsHTMLWhitespace(aReferenceString.Last()));
   uint32_t referenceStringPosition =
       direction == TextScanDirection::Left ? aReferenceString.Length() - 1 : 0;
 